### PR TITLE
minizincide: 2.5.5 -> 2.7.6

### PR DESCRIPTION
**Pull request by Vinetos**
## Description of changes

MinizinIDE was unable to be launched because it cannot found plugins
image

This MR update MinizincIDE the to latest version and add `QT_QPA_PLATFORM_PLUGIN_PATH` variable to the wrapper

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Vinetos** [comment]
@dtzWill May I have a review from you ?

**Comment by Vinetos** [comment]
Result of `nixpkgs-review pr 265073` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>minizincide</li>
  </ul>
</details>

**Comment by Vinetos** [comment]
@ofborg eval

File: pkgs/development/tools/minizinc/ide.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "minizinc-ide";
-  version = "2.5.5";
+  version = "2.7.6";
 
   src = fetchFromGitHub {
     owner = "MiniZinc";
     repo = "MiniZincIDE";
     rev = version;
-    sha256 = "sha256-0U3KFRDam8psbCaEOcrwqzICAy1oBgo8SFEiR/PMqZk=";
+    sha256 = "sha256-NUGQmOkdnEZq+wr/VhjSR/IBwF565U4QlTkmpjOYJes=";
```
Comment (liff) [inline pkgs/development/tools/minizinc/ide.nix:11]: Applied against master and building, I get this error (after enabling NIXPKGS_ALLOW_INSECURE due to QtWebKit):

```
source> removing `.git'...
error: hash mismatch in fixed-output derivation '/nix/store/fv24ifwcsw9cc3gawr4wfzanlhbi8v7f-source.drv':
         specified: sha256-NUGQmOkdnEZq+wr/VhjSR/IBwF565U4QlTkmpjOYJes=
            got:    sha256-Fwt4zuIKwurCx0EriwSOea4zAJ7Bmr8Stt0dQ8Ux0cs=
error: 1 dependencies of derivation '/nix/store/z1wqw3gy3d6cfqs80lzv0pjqsdw8njqz-minizinc-ide-2.7.6.drv' failed to build
```

Comment (Vinetos) [reply pkgs/development/tools/minizinc/ide.nix]: Should be fixed

**Comment by liff** [comment]
Result of `nixpkgs-review pr 265073` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>minizincide</li>
  </ul>
</details>

**Comment by liff** [comment]
Here’s the log from the build failure:

```
building '/nix/store/4d9s0aira9q38ss3vz7y921mwxyhc55f-minizinc-ide-2.7.6.drv'...
qmakePrePhase
qtPreHook
unpacking sources
unpacking source archive /nix/store/ji1ahj8s83y014gv7q9j0s378cfiz6zb-source
source root is source/MiniZincIDE
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
QMAKEPATH=/nix/store/fawjz5gzrjs6jb7vs7p0fhx65wrv6zwh-qtbase-5.15.11-dev:/nix/store/a627aiw1awwc8lzgmkqj7ffwghy3f2q0-qtwayland-5.15.11-dev:/nix/store/lbk8r8g5f90hqm7cpbj6p9bs8468rwyb-qtdeclarative-5.15.11-dev:/nix/store/shqyklq66zhmz40296nz2ymacanmjwf1-qtsvg-5.15.11-dev:/nix/store/wnxy0nppz9k1r61xqamg7mbph8609anr-qtwebengine-5.15.15-dev:/nix/store/a92dr2ngzgl7fm1rsvniwm6wwcqipqqq-qtlocation-5.15.11-dev:/nix/store/0w7k0692v1dx4sb90phh39jfhdjcx95h-qtmultimedia-5.15.11-dev:/nix/store/0w217m58w6hy88a2pad7v7h02719fpr9-qtwebchannel-5.15.11-dev:/nix/store/wzw1hm2ygkl24w3x6h5di50v21v4m2w6-qtwebkit-5.212.0-alpha4-dev:/nix/store/m87h6gaazg8vylw9n3vadv7am8ris92g-qtsensors-5.15.11-dev
qmake PREFIX=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6 NIX_OUTPUT_OUT=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6 NIX_OUTPUT_DEV=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6 NIX_OUTPUT_BIN=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6 NIX_OUTPUT_DOC=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6/share/doc/qt-5.15.11 NIX_OUTPUT_QML=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6/lib/qt-5.15.11/qml NIX_OUTPUT_PLUGIN=/nix/store/cs151vxqpgsgvyq24kad418g9lsmggmr-minizinc-ide-2.7.6/lib/qt-5.15.11/plugins CONFIG+=release CONFIG+=nostrip
Info: creating stash file /build/source/MiniZincIDE/.qmake.stash
Project ERROR: Unknown module(s) in QT: websockets
error: builder for '/nix/store/4d9s0aira9q38ss3vz7y921mwxyhc55f-minizinc-ide-2.7.6.drv' failed with exit code 3;
```

**Comment by Dekker1** [comment]
It seems like this PR has been stale for quite a while, but I really would hope that the MiniZincIDE would be updated soon.

Today, I actually added a flake to the MiniZincIDE, that uses the following package definition for MiniZinc 2.8.2, that seems to work correctly.

```nix
{ self, lib, stdenv, qt6, minizinc, makeWrapper, Cocoa }:

stdenv.mkDerivation rec {
  name = "minizinc-ide";

  src = self;
  nativeBuildInputs = [ qt6.qmake makeWrapper ];
  buildInputs = [ qt6.qtbase qt6.qtwebsockets ] ++ lib.optionals stdenv.isDarwin [ Cocoa ];
  dontWrapQtApps = true;

  executableLoc = if stdenv.isDarwin then "$out/bin/MiniZincIDE.app/Contents/MacOS/MiniZincIDE" else "$out/bin/MiniZincIDE";
  postInstall = ''
    wrapProgram ${executableLoc} --prefix PATH ":" ${lib.makeBinPath [ minizinc ]} --set QT_QPA_PLATFORM_PLUGIN_PATH "${lib.makeBinPath [ qt6.qtbase ]}/../lib/qt-6/plugins/platforms"
  '';

  meta = with lib; {
    homepage = "https://www.minizinc.org/";
    description = "IDE for MiniZinc, a medium-level constraint modelling language";
    longDescription = ''
      MiniZinc is a medium-level constraint modelling
      language. It is high-level enough to express most
      constraint problems easily, but low-level enough
      that it can be mapped onto existing solvers easily and consistently.
      It is a subset of the higher-level language Zinc.
    '';
    license = licenses.mpl20;
    platforms = platforms.unix;
  };
}
```

Note that it now also works on MacOS and uses the correct Qt plugins

**Comment by Vinetos** [comment]
Thanks ! I will give it a look soon

**Comment by Vinetos** [comment]
Replaced by https://github.com/NixOS/nixpkgs/pull/284990

**Pull request closed without merging**